river/View.zig

```diff
@@ -488,14 +488,65 @@ pub fn map(view: *Self) !void {
     if (server.config.float_rules.match(view)) |float| {
         view.pending.float = float;
     }
+    if (server.config.fullscreen_rules.match(view)) |fullscreen| {
+        view.pending.fullscreen = fullscreen;
+    }
     if (server.config.ssd_rules.match(view)) |ssd| {
         view.pending.ssd = ssd;
     }
 
     if (server.input_manager.defaultSeat().focused_output) |output| {
-        // Center the initial pending box on the output
-        view.pending.box.x = @divTrunc(@max(0, output.usable_box.width - view.pending.box.width), 2);
-        view.pending.box.y = @divTrunc(@max(0, output.usable_box.height - view.pending.box.height), 2);
+        const border_width = if (view.pending.ssd) server.config.border_width else 0;
+
+        if (server.config.position_rules.match(view)) |position| {
+            // Set the view position clamping to the bounds of the output.
+            var out_of_bounds = false;
+
+            const max_x = output.usable_box.width - border_width - view.pending.box.width;
+            if (position.x < border_width or position.x > max_x) out_of_bounds = true;
+            view.pending.box.x = position.x;
+            view.pending.box.x = @max(view.pending.box.x, border_width);
+            view.pending.box.x = @min(view.pending.box.x, max_x);
+            view.pending.box.x = @max(view.pending.box.x, 0);
+
+            const max_y = output.usable_box.height - border_width - view.pending.box.height;
+            if (position.y < border_width or position.y > max_y) out_of_bounds = true;
+            view.pending.box.y = position.y;
+            view.pending.box.y = @max(view.pending.box.y, border_width);
+            view.pending.box.y = @min(view.pending.box.y, max_y);
+            view.pending.box.y = @max(view.pending.box.y, 0);
+
+            if (out_of_bounds) {
+                log.warn("view '{?s}' position {d},{d} out of bounds, fallback to default", .{
+                    view.getTitle(),
+                    position.x,
+                    position.y,
+                });
```

Comment:
Instead of duplicating all this bounds checking logic here I suggest the following:
1. Set the pending position of the view to a matching position rule or center it.
2. Set the pending dimensions of the view to a matching dimensions rule if present.
3. That's it, you're done. The view.setPendingOutput() call below will clamp the view's position/dimensions to the constraints and output dimensions.

There's no need to do any logging about the requested dimensions being out of bounds either IMO.

Reply:
The reason this whole block exists is solely because of the need to check the out of bounds condition before setting the position/dimensions of the view. My original thought process was that since the behavior of the position and dimensions rule are dependent on the output dimensions, I should log when the behavior changes. If this is deemed unnecessary, I can remove the block and use the suggested `view.setPendingOutput()` approach.

Reply:
As I said in my original comment, I don't think there is a need to do any logging in the case that the position/dimensions place the view out of bounds. Thanks for cleaning things up!

Reply:
Alright, done.